scipy/stats/_survival.py

```diff
@@ -374,3 +391,138 @@ def _ecdf_right_censored(sample):
     sf = np.cumprod((n - d) / n)
     cdf = 1 - sf
     return t, cdf, sf, n, d
+
+
+@dataclass
+class LogRankResult:
+    statistic: np.ndarray
+    pvalue: np.ndarray
+
+
+def log_rank(
+    x: npt.ArrayLike | CensoredData,
+    y: npt.ArrayLike | CensoredData
+) -> LogRankResult:
+    """Compare the survival distributions of two samples via the logrank test.
+
+    Parameters
+    ----------
+    x, y : array_like or CensoredData
+        Samples to compare based on their survival functions.
+
+    Returns
+    -------
+    res : LogRankResult
+        An object containing attributes:
+
+        statistic : float ndarray
+            The computed statistic of the test.
+        pvalue : float ndarray
+            The computed p-value of the test.
+
+    See Also
+    --------
+    scipy.stats.ecdf
+
```

Comment:
Please add notes about which version of the test is performed.

Reply:
I added some notes. Hopefully it's clear and would help also when reading the code comments.